lib/task_005.rb

```diff
@@ -48,55 +49,20 @@
 module Task005
   module_function
 
-  def parse_input(input_data)
-    input_data.split("\n").map { |line| parse_line(line) }
-  end
-
-  def parse_line(line)
-    point1, point2 = line.split(' -> ')
-
-    {
-      point1: parse_point(point1),
-      point2: parse_point(point2),
-    }
-  end
-
-  def parse_point(point)
-    point.split(',').map { |value| value.to_i }
-  end
-
-  def mark_lines(lines)
-    diagram = {}
-    lines.each { |line| mark_line(line, diagram) }
-    diagram
-  end
-
-  def mark_line(line, diagram)
-    if line[:point1][0] == line[:point2][0] || line[:point1][1] == line[:point2][1]
-      x1, x2 = [line[:point1][0], line[:point2][0]].sort
-      y1, y2 = [line[:point1][1], line[:point2][1]].sort
-      (x1..x2).each do |x|
-        (y1..y2).each do |y|
-          diagram.key?([x, y]) ? diagram[[x, y]] += 1 : diagram.store([x, y], 1)
-        end
-      end
-    end
-    diagram
-  end
-
-  def calculate_answer(input_data)
+  def calculate_answer_005(input_data)
```

Comment:
here I use one module for task_005 and task_005a, becouse they use one lib file.

Reply:
Hmm ok, I didn't notice that. Then I would call it `calculate_answer_for_part_1` and `calculate_answer_for_part_2` or something similar. I mean, it is already `005` module, so `005` does not add anything new I guess. But maybe it is not that important.

> they use one lib file.
The `lib` file is not something that people would do in Ruby.

Usually it should be structured like this:

```
|-- task005.rb
|-- task0055/
    |-- point.rb
    |-- line.rb
    |-- diagram.rb
```

where `task005.rb` contains `Task005` module (it could have 2 functions: `Task005.calculate_answer_for_part_1` and Task005.calculate_answer_for_part_2`). And if you have any submodule, or class that goes inside `Task005` module, you should put it into the `task0055` directory.

Reply:
it is changed already with the names, but maybe I should do `task005` directory and each class to separate files like you suggest? do you think that is good idea for this project or pull back classes to the `task_005.rb` file?

Reply:
another question is: could one module be in 2 files? like in example: `task_005.rb` and `diagram.rb` (currently Module5 is in task_005.rb and task_005_lib.rb

Reply:
I noticed we are starting to solve two problems at the time:
- refactor task 5 to use object oriented programming
- how to share the code between part 1 and part 2 of the same task

I think we should do a step back and focus on one problem at the time. I would suggest leaving it as it is in this PR. And if you feel like it, we can deal with this issue but maybe as part of a separate PR.

One of the important skills in software development (probably the same as in many other areas) is creating small PR (focusing on one problem at a time) ;)